### PR TITLE
[Workers] Version previews behind Access

### PR DESCRIPTION
**Pull request by tanushree-sharma**
Looks like the next.js note was already fixed: https://developers.cloudflare.com/workers/frameworks/framework-guides/nextjs/

Just need reviews on the previews section.

### CODE REVIEW

File: src/content/docs/workers/configuration/previews.mdx [hunk 1]
```diff
@@ -12,7 +12,7 @@ import { Render } from "~/components";
 
 Preview URLs allow you to preview new versions of your Worker without deploying it to production.
 
-Every time you create a new [version](/workers/configuration/versions-and-deployments/#versions) of your Worker a unique preview URL is generated. New [versions](/workers/configuration/versions-and-deployments/#versions) of a Worker are created on [`wrangler deploy`](/workers/wrangler/commands/#deploy), [`wrangler versions upload`](/workers/wrangler/commands/#upload) or when you make edits on the Cloudflare dashboard. By default, preview URLs are enabled and available publicly.
+Every time you create a new [version](/workers/configuration/versions-and-deployments/#versions) of your Worker a unique preview URL is generated. Preview URLs take the format: `versionPrefix.workerName.subdomainName.workers.dev`. New [versions](/workers/configuration/versions-and-deployments/#versions) of a Worker are created on [`wrangler deploy`](/workers/wrangler/commands/#deploy), [`wrangler versions upload`](/workers/wrangler/commands/#upload) or when you make edits on the Cloudflare dashboard. By default, preview URLs are enabled and available publicly.
```
Comment (GregBrimble) [inline src/content/docs/workers/configuration/previews.mdx]: Generally, you want to distinguish between variables and any fixed content in concatenated strings like this.

Also, I thought it was `<VERSION_PREFIX>-<WORKER_NAME>.<SUBDOMAIN>.workers.dev` not `<VERSION_PREFIX>.<WORKER_NAME>.<SUBDOMAIN>.workers.dev`??

```suggestion
Every time you create a new [version](/workers/configuration/versions-and-deployments/#versions) of your Worker a unique preview URL is generated. Preview URLs take the format: `<VERSION_PREFIX>.<WORKER_NAME>.<SUBDOMAIN>.workers.dev`. New [versions](/workers/configuration/versions-and-deployments/#versions) of a Worker are created on [`wrangler deploy`](/workers/wrangler/commands/#deploy), [`wrangler versions upload`](/workers/wrangler/commands/#upload) or when you make edits on the Cloudflare dashboard. By default, preview URLs are enabled and available publicly.
```

File: src/content/docs/workers/configuration/previews.mdx [hunk 2]
```diff
@@ -35,6 +37,33 @@ The [`wrangler versions upload`](/workers/wrangler/commands/#upload) command upl
 1. Log in to the [Cloudflare dashboard](https://dash.cloudflare.com/?to=/:account/workers) and select your project.
 2. Head to the "Deployments" tab find the version you would like to view.
 
+## Manage access to Preview URLs
+
+By default, preview URLs are enabled and available publicly. You can use [Cloudflare Access](/cloudflare-one/policies/access/) to require visitors to authenticate before accessing preview URLs. You can limit access to yourself, your teammates, your organization, or anyone else you specify in your [access policy](/cloudflare-one/policies/access).
+
+To limit your preview URLs to authorized emails only:
+
+1. Log in to the [Cloudflare Access dashboard](https://one.dash.cloudflare.com/?to=/:account/access/apps)
```
Comment (dcpena) [inline src/content/docs/workers/configuration/previews.mdx]:
```suggestion
1. Log in to the [Cloudflare Access dashboard](https://one.dash.cloudflare.com/?to=/:account/access/apps).
```

File: src/content/docs/workers/configuration/previews.mdx [hunk 3]
```diff
@@ -35,6 +37,33 @@ The [`wrangler versions upload`](/workers/wrangler/commands/#upload) command upl
...
+2. Select your account
```
Comment (dcpena) [inline src/content/docs/workers/configuration/previews.mdx]:
```suggestion
2. Select your account.
```

File: src/content/docs/workers/configuration/previews.mdx [hunk 4]
```diff
@@ -24,6 +24,8 @@ When testing zone level performance or security featues for a version, we recomm
 
 :::note
 Preview URLs are only available for Worker versions uploaded after 2024-09-25.
+
+Minimum required Wrangler version: 3.74.0. Check your version by running `wrangler version`. To update Wrangler, refer to [Install/Update Wrangler](/workers/wrangler/install-and-update/).
```
Comment (GregBrimble) [inline src/content/docs/workers/configuration/previews.mdx]: `wrangler version` is deprecated.

```suggestion
Minimum required Wrangler version: 3.74.0. Check your version by running `wrangler --version`. To update Wrangler, refer to [Install/Update Wrangler](/workers/wrangler/install-and-update/).
```

File: src/content/docs/workers/configuration/previews.mdx [hunk 5]
```diff
@@ -35,6 +37,33 @@ The [`wrangler versions upload`](/workers/wrangler/commands/#upload) command upl
...
 
+## Manage access to Preview URLs
+
+By default, preview URLs are enabled and available publicly. You can use [Cloudflare Access](/cloudflare-one/policies/access/) to require visitors to authenticate before accessing preview URLs. You can limit access to yourself, your teammates, your organization, or anyone else you specify in your [access policy](/cloudflare-one/policies/access).
+
+To limit your preview URLs to authorized emails only:
+
+1. Log in to the [Cloudflare Access dashboard](https://one.dash.cloudflare.com/?to=/:account/access/apps)
+2. Select your account
+3. Add an application
```
Comment (dcpena) [inline src/content/docs/workers/configuration/previews.mdx]:
```suggestion
3. Add an application.
```

File: src/content/docs/workers/configuration/previews.mdx [hunk 6]
```diff
@@ -35,6 +37,33 @@ The [`wrangler versions upload`](/workers/wrangler/commands/#upload) command upl
...
+4. Select "Self Hosted"
+5. Name your application (for example, My-Worker) and add your `workers.dev` subdomain as the "Application domain".
```
Comment (GregBrimble) [inline src/content/docs/workers/configuration/previews.mdx]:
```suggestion
5. Name your application (for example, "my-worker") and add your `workers.dev` subdomain as the "Application domain".
```

File: src/content/docs/workers/configuration/previews.mdx [hunk 7]
```diff
@@ -35,6 +37,33 @@ The [`wrangler versions upload`](/workers/wrangler/commands/#upload) command upl
...
+## Manage access to Preview URLs
+
+By default, preview URLs are enabled and available publicly. You can use [Cloudflare Access](/cloudflare-one/policies/access/) to require visitors to authenticate before accessing preview URLs. You can limit access to yourself, your teammates, your organization, or anyone else you specify in your [access policy](/cloudflare-one/policies/access).
+
+To limit your preview URLs to authorized emails only:
+
+1. Log in to the [Cloudflare Access dashboard](https://one.dash.cloudflare.com/?to=/:account/access/apps)
+2. Select your account
+3. Add an application
+4. Select "Self Hosted"
```
Comment (dcpena) [inline src/content/docs/workers/configuration/previews.mdx]:
```suggestion
4. Select **Self Hosted**.
```

File: src/content/docs/workers/configuration/previews.mdx [hunk 8]
```diff
@@ -35,6 +37,33 @@ The [`wrangler versions upload`](/workers/wrangler/commands/#upload) command upl
...
+5. Name your application (for example, My-Worker) and add your `workers.dev` subdomain as the "Application domain".
+
+For example, if you want to secure preview URLs for a Worker running on `my-worker.my-subdomain.workers.dev`.
+
+- Subdomain: `*.my-worker`
```
Comment (GregBrimble) [inline src/content/docs/workers/configuration/previews.mdx]: Again, I thought it was `*-my-worker`? Did this change?

File: src/content/docs/workers/configuration/previews.mdx (hunk 6 above)
Comment (dcpena) [inline src/content/docs/workers/configuration/previews.mdx]:
```suggestion
5. Name your application (for example, `My-Worker`) and add your `workers.dev` subdomain as the **Application domain**.
```

File: src/content/docs/workers/configuration/previews.mdx [hunk 9]
```diff
@@ -35,6 +37,33 @@ The [`wrangler versions upload`](/workers/wrangler/commands/#upload) command upl
...
+- Domain: `my-subdomain.workers.dev`
+
+:::note
+You must press enter after you input your Application domain for it to save. You will see a "Zone is not associated with the current account" warning that you may ignore.
+:::
+
+6. Go to the next page
+7. Add a name for your access policy (for example, Authorized preview emails for my-Worker)
```
Comment (GregBrimble) [inline src/content/docs/workers/configuration/previews.mdx]:
```suggestion
7. Add a name for your access policy (for example, "Allow employees access to preview URLs for my-worker")
```

File: src/content/docs/workers/configuration/previews.mdx (hunk 6 above)
Comment (tanushree-sharma) [reply src/content/docs/workers/configuration/previews.mdx]: ugh AI linter. should have checked more carefully

File: src/content/docs/workers/configuration/previews.mdx [hunk 10]
```diff
@@ -35,6 +37,33 @@ The [`wrangler versions upload`](/workers/wrangler/commands/#upload) command upl
...
+8. In the "Configure rules" section create a new rule with the "Emails" selector. Enter the emails you want to authorize to access the version preview. View [access policies](/cloudflare-one/policies/access/#selectors) to learn about configuring alternate rules.
```
Comment (GregBrimble) [inline src/content/docs/workers/configuration/previews.mdx]:
```suggestion
8. In the "Configure rules" section create a new rule with the "Emails" selector, or any other attributes which you wish to gate access to previews with. Enter the emails you want to authorize. View [access policies](/cloudflare-one/policies/access/#selectors) to learn about configuring alternate rules.
```

File: src/content/docs/workers/configuration/previews.mdx (hunk 1 above)
Comment (tanushree-sharma) [reply src/content/docs/workers/configuration/previews.mdx]: You're right on format. Fixing

File: src/content/docs/workers/frameworks/framework-guides/nextjs.mdx [hunk 11]
```diff
@@ -23,8 +23,10 @@ npm create cloudflare@latest --no-auto-update --experimental --framework next
 
 #### Existing Next.js apps
 
-::: note
+:::note
+
 Minimum required Wrangler version: 3.78.10. Check your version by running `wrangler version`. To update Wrangler, refer to [Install/Update Wrangler](/workers/wrangler/install-and-update/).
```
Comment (GregBrimble) [inline src/content/docs/workers/frameworks/framework-guides/nextjs.mdx]: `wrangler version` is deprecated.

```suggestion
Minimum required Wrangler version: 3.78.10. Check your version by running `wrangler --version`. To update Wrangler, refer to [Install/Update Wrangler](/workers/wrangler/install-and-update/).
```

**Review by GregBrimble — APPROVED** [review]

File: src/content/docs/workers/configuration/previews.mdx [hunk 12]
```diff
@@ -35,6 +37,33 @@ The [`wrangler versions upload`](/workers/wrangler/commands/#upload) command upl
...
+
+1. Log in to the [Cloudflare Access dashboard](https://one.dash.cloudflare.com/?to=/:account/access/apps)
+2. Select your account
+3. Add an application
+4. Select "Self Hosted"
+5. Name your application (for example, "my-worker") and add your `workers.dev` subdomain as the "Application domain".
+
+For example, if you want to secure preview URLs for a Worker running on `my-worker.my-subdomain.workers.dev`.
+
+- Subdomain: `*.my-worker`
```
Comment (tanushree-sharma) [inline src/content/docs/workers/configuration/previews.mdx]:
```suggestion
- Subdomain: `*-my-worker`
```

File: src/content/docs/workers/configuration/previews.mdx (hunk 1 above)
Comment (tanushree-sharma) [inline src/content/docs/workers/configuration/previews.mdx:15]: Every time you create a new [version](/workers/configuration/versions-and-deployments/#versions) of your Worker a unique preview URL is generated. Preview URLs take the format: `<VERSION_PREFIX>-<WORKER_NAME>.<SUBDOMAIN>.workers.dev`. New [versions](/workers/configuration/versions-and-deployments/#versions) of a Worker are created on [`wrangler deploy`](/workers/wrangler/commands/#deploy), [`wrangler versions upload`](/workers/wrangler/commands/#upload) or when you make edits on the Cloudflare dashboard. By default, preview URLs are enabled and available publicly.

Comment (tanushree-sharma) [inline src/content/docs/workers/configuration/previews.mdx]:
```suggestion
Every time you create a new [version](/workers/configuration/versions-and-deployments/#versions) of your Worker a unique preview URL is generated. Preview URLs take the format: `<VERSION_PREFIX>-<WORKER_NAME>.<SUBDOMAIN>.workers.dev`. New [versions](/workers/configuration/versions-and-deployments/#versions) of a Worker are created on [`wrangler deploy`](/workers/wrangler/commands/#deploy), [`wrangler versions upload`](/workers/wrangler/commands/#upload) or when you make edits on the Cloudflare dashboard. By default, preview URLs are enabled and available publicly.
```

File: src/content/docs/workers/configuration/previews.mdx [hunk 13]
```diff
@@ -35,6 +37,33 @@ The [`wrangler versions upload`](/workers/wrangler/commands/#upload) command upl
...
+
+To limit your preview URLs to authorized emails only:
+
+1. Log in to the [Cloudflare Access dashboard](https://one.dash.cloudflare.com/?to=/:account/access/apps)
+2. Select your account
+3. Add an application
+4. Select "Self Hosted"
+5. Name your application (for example, My-Worker) and add your `workers.dev` subdomain as the "Application domain".
+
+For example, if you want to secure preview URLs for a Worker running on `my-worker.my-subdomain.workers.dev`.
```
Comment (dcpena) [inline src/content/docs/workers/configuration/previews.mdx:52]: This reads a bit incomplete. It seems like it should say "if you want to secure preview URLs, do x" but that latter part is missing.

File: src/content/docs/workers/configuration/previews.mdx [hunk 14]
```diff
@@ -35,6 +37,33 @@ The [`wrangler versions upload`](/workers/wrangler/commands/#upload) command upl
...
+
+- Subdomain: `*.my-worker`
+- Domain: `my-subdomain.workers.dev`
+
+:::note
+You must press enter after you input your Application domain for it to save. You will see a "Zone is not associated with the current account" warning that you may ignore.
+:::
+
+6. Go to the next page
```
Comment (dcpena) [inline src/content/docs/workers/configuration/previews.mdx]:
```suggestion
6. Go to the next page.
```

File: src/content/docs/workers/configuration/previews.mdx (hunk 9 above)
Comment (dcpena) [inline src/content/docs/workers/configuration/previews.mdx]:
```suggestion
7. Add a name for your access policy (for example, Authorized preview emails for my-Worker).
```

File: src/content/docs/workers/configuration/previews.mdx (hunk 10 above)
Comment (dcpena) [inline src/content/docs/workers/configuration/previews.mdx]:
```suggestion
8. In the **Configure rules** section, create a new rule with the **Emails** selector. 
9. Enter the emails you want to authorize to access the version preview. View [access policies](/cloudflare-one/policies/access/#selectors) to learn about configuring alternate rules.
```

File: src/content/docs/workers/configuration/previews.mdx [hunk 15]
```diff
@@ -35,6 +37,33 @@ The [`wrangler versions upload`](/workers/wrangler/commands/#upload) command upl
...
+7. Add a name for your access policy (for example, Authorized preview emails for my-Worker)
+8. In the "Configure rules" section create a new rule with the "Emails" selector. Enter the emails you want to authorize to access the version preview. View [access policies](/cloudflare-one/policies/access/#selectors) to learn about configuring alternate rules.
+9. Go to the next page
```
Comment (dcpena) [inline src/content/docs/workers/configuration/previews.mdx]:
```suggestion
10. Go to the next page.
```

File: src/content/docs/workers/configuration/previews.mdx [hunk 16]
```diff
@@ -35,6 +37,33 @@ The [`wrangler versions upload`](/workers/wrangler/commands/#upload) command upl
...
+10. Add application
```
Comment (dcpena) [inline src/content/docs/workers/configuration/previews.mdx]:
```suggestion
11. Add application.
```

**Review by irvinebroque — APPROVED** [review]

File: src/content/docs/workers/configuration/previews.mdx [hunk 17]
```diff
@@ -35,6 +37,33 @@ The [`wrangler versions upload`](/workers/wrangler/commands/#upload) command upl
...
+- Domain: `my-subdomain.workers.dev`
+
+:::note
+You must press enter after you input your Application domain for it to save. You will see a "Zone is not associated with the current account" warning that you may ignore.
+:::
+
+6. Go to the next page
+7. Add a name for your access policy (for example, "Allow employees access to preview URLs for my-worker")
+8. In the "Configure rules" section create a new rule with the "Emails" selector, or any other attributes which you wish to gate access to previews with. Enter the emails you want to authorize. View [access policies](/cloudflare-one/policies/access/#selectors) to learn about configuring alternate rules.
+9. Go to the next page
```
Comment (tanushree-sharma) [inline src/content/docs/workers/configuration/previews.mdx]:
```suggestion
9. Go to the next page.
```

File: src/content/docs/workers/configuration/previews.mdx [hunk 18]
```diff
@@ -35,6 +37,33 @@ The [`wrangler versions upload`](/workers/wrangler/commands/#upload) command upl
...
+10. Add application
```
Comment (tanushree-sharma) [inline src/content/docs/workers/configuration/previews.mdx]:
```suggestion
10. Add application.
```

File: src/content/docs/workers/configuration/previews.mdx [hunk 19]
```diff
@@ -35,6 +37,33 @@ The [`wrangler versions upload`](/workers/wrangler/commands/#upload) command upl
...
+
+By default, preview URLs are enabled and available publicly. You can use [Cloudflare Access](/cloudflare-one/policies/access/) to require visitors to authenticate before accessing preview URLs. You can limit access to yourself, your teammates, your organization, or anyone else you specify in your [access policy](/cloudflare-one/policies/access).
+
+To limit your preview URLs to authorized emails only:
+
+1. Log in to the [Cloudflare Access dashboard](https://one.dash.cloudflare.com/?to=/:account/access/apps).
+2. Select your account.
+3. Add an application.
+4. Select **Self Hosted**.
+5. Name your application (for example, "my-worker") and add your `workers.dev` subdomain as the "Application domain".
```
Comment (tanushree-sharma) [inline src/content/docs/workers/configuration/previews.mdx]:
```suggestion
5. Name your application (for example, "my-worker") and add your `workers.dev` subdomain as the **Application domain**.
```

File: src/content/docs/workers/configuration/previews.mdx [hunk 20]
```diff
@@ -35,6 +37,33 @@ The [`wrangler versions upload`](/workers/wrangler/commands/#upload) command upl
...
+
+- Subdomain: `*-my-worker`
+- Domain: `my-subdomain.workers.dev`
+
+:::note
+You must press enter after you input your Application domain for it to save. You will see a "Zone is not associated with the current account" warning that you may ignore.
+:::
+
+6. Go to the next page.
+7. Add a name for your access policy (for example, "Allow employees access to preview URLs for my-worker")
```
Comment (tanushree-sharma) [inline src/content/docs/workers/configuration/previews.mdx]:
```suggestion
7. Add a name for your access policy (for example, "Allow employees access to preview URLs for my-worker").
```

File: src/content/docs/workers/configuration/previews.mdx [hunk 21]
```diff
@@ -35,6 +37,33 @@ The [`wrangler versions upload`](/workers/wrangler/commands/#upload) command upl
...
+8. In the "Configure rules" section create a new rule with the "Emails" selector, or any other attributes which you wish to gate access to previews with. Enter the emails you want to authorize. View [access policies](/cloudflare-one/policies/access/#selectors) to learn about configuring alternate rules.
```
Comment (tanushree-sharma) [inline src/content/docs/workers/configuration/previews.mdx]:
```suggestion
8. In the **Configure rules** section create a new rule with the **Emails** selector, or any other attributes which you wish to gate access to previews with. 
9. Enter the emails you want to authorize. View [access policies](/cloudflare-one/policies/access/#selectors) to learn about configuring alternate rules.
```

File: src/content/docs/workers/configuration/previews.mdx [hunk 22]
```diff
@@ -35,6 +37,33 @@ The [`wrangler versions upload`](/workers/wrangler/commands/#upload) command upl
...
+9. Go to the next page.
```
Comment (tanushree-sharma) [inline src/content/docs/workers/configuration/previews.mdx]:
```suggestion
10. Go to the next page.
```

File: src/content/docs/workers/configuration/previews.mdx [hunk 23]
```diff
@@ -35,6 +37,33 @@ The [`wrangler versions upload`](/workers/wrangler/commands/#upload) command upl
...
+10. Add application.
```
Comment (tanushree-sharma) [inline src/content/docs/workers/configuration/previews.mdx]:
```suggestion
11. Add application.
```

**Pull request merged**